lint/rule_validate.go

```diff
@@ -0,0 +1,105 @@
+package lint
+
+import (
+	"sync"
+
+	"cuelang.org/go/cue/errors"
+	"github.com/grafana/grafana/pkg/schema"
+	"github.com/grafana/grafana/pkg/schema/load"
+)
+
+// Performance-guarding logic for loading the Grafana dashboard scuemata.
+//
+// We use these because the CUE runtime currently (as of v0.4.0) has performance issues with
+// large disjunctions, which are a necessary part of how the "dist" dashboard schema composes
+// in schema from Grafana plugins. These issues only manifest on initial load, so we a) do not
+// do the load in an init() function, and b) guard it with a sync.Once in order to ensure that
+// the cost is paid only once even for multiple calls.
+var baseOnce, distOnce sync.Once
+var basesch, distsch schema.CueSchema
+
+func getBaseSchema() (schema.CueSchema, error) {
+	var err error
```

Comment:
Bleh, this means only the first call will actually error correctly, defeating the point of a Once.

If we actually want to merge this, i'll fix the problem.